tests/PHPStan/Analyser/AnalyserIntegrationTest.php

```diff
@@ -601,6 +601,13 @@ public function testBug6649(): void
 		$this->assertNoErrors($errors);
 	}
 
+
+	public function testBug6889(): void
+	{
+		$errors = $this->runAnalyse(__DIR__ . '/data/bug-6889.php');
```

Comment:
Instead of AnalyserIntegrationTest, just put this file in NodeScopeResolverTest instead.